File: typescript/src/ML/easy/valid_anagram.ts
export function isAnagram(s: string, t: string): boolean {
  if (s.length != t.length) return false;
  const charCounts = new Array(26).fill(0);
  for (let i = 0; i < s.length; i++) {
    charCounts[s.charCodeAt(i) - "a".charCodeAt(0)]++;
    charCounts[t.charCodeAt(i) - "a".charCodeAt(0)]--;
  }
  for (let count of charCounts) {
    if (count != 0) return false;
  }
  return true;
}
function isAnagram1(s: string, t: string): boolean {
  if (s.length != t.length) return false;
  const charCounts = new Array(26).fill(0);
  for (let i = 0; i < s.length; i++) {
    charCounts[s.charCodeAt(i) - "a".charCodeAt(0)]++;
    charCounts[t.charCodeAt(i) - "a".charCodeAt(0)]--;
  }
  return charCounts.every((char) => char === 0);
}
function isAnagram2(s: string, t: string): boolean {
  return s.split("").sort().join("") === t.split("").sort().join("");
}

File: python/src/ML/easy/valid_anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        if len(s)!= len(t):
            return False
        char_count = [0] * 26
        for i in range(0,len(s)):
            char_count[ord(s[i])-ord('a')]+=1
            char_count[ord(t[i])-ord('a')]-=1
        for count in char_count:
            if count!= 0:
                return False
        return True
    def isAnagram1(self, s: str, t: str) -> bool:
        if len(s)!= len(t):
            return False
        char_count = [0] * 26
        for i in range(0,len(s)):
            char_count[ord(s[i])-ord('a')]+=1
            char_count[ord(t[i])-ord('a')]-=1
        return all(count== 0 for count in char_count)